src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,121 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int capacity = DEFAULT_CAPACITY;
+    private Node<K, V>[] table = new Node[capacity];
+    private int size = 0;
+    private int threshold = 0;
 
     @Override
     public void put(K key, V value) {
+        int hash = key == null ? 0 : Math.abs(key.hashCode());
+        int index = Math.abs(hash) % capacity;
+        if (index < 0) {
+            index += capacity;
+        }
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+
+        Node<K, V> newNode = new Node<>(key, value);
```

Comment:
The Node class is parameterized, but you're not using the diamond operator `<>` when creating a new Node instance. It's better to use it to avoid unchecked assignment warnings.